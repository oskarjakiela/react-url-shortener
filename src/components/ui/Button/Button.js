import classnames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';

import './Button.css';


const displayName = 'Shorty-Button';

const Button = ({
  children,
  className,
  ...props
}) => (
  <button className={classnames(displayName, className, {
    [`${displayName}--disabled`]: props.disabled,
  })} {...props}>
    {children}
  </button>
);

Button.displayName = displayName;

Button.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Button;
