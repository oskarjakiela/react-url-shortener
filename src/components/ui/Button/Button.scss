.Shorty-Button {
  display: inline-block;
  border: 0;
  border-radius: 3px;
  padding: 0 1.125rem;
  height: 2.625rem;
  font-family: 'Roboto', sans-serif;
  font-weight: 300;
  font-size: 1rem;
  text-align: center;
  text-decoration: none;
  background-color: #eb4a42;
  color: #fff;
  white-space: nowrap;
  vertical-align: middle;
  cursor: pointer;
  user-select: none;
  outline: none;

  &--disabled {
    background-color: #e0e0e0;
    color: #bfbfbf;
  }
}

