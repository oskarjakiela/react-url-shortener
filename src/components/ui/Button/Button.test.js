import React from 'react';
import renderer from 'react-test-renderer';

import Button from './Button';


it('<Button /> renders correctly', () => {
  const tree = renderer
    .create(<Button>Lorem ipsum</Button>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('<Button className="foo-bar" /> renders correctly', () => {
  const tree = renderer
    .create(<Button className="foo-bar">Lorem ipsum</Button>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('<Button disabled /> renders correctly', () => {
  const tree = renderer
    .create(<Button disabled>Lorem ipsum</Button>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
