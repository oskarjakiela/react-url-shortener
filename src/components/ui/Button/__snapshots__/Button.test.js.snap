// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly 1`] = `
<button
  className="Shorty-Button"
>
  Lorem ipsum
</button>
`;

exports[`<Button className="foo-bar" /> renders correctly 1`] = `
<button
  className="Shorty-Button foo-bar"
>
  Lorem ipsum
</button>
`;

exports[`<Button disabled /> renders correctly 1`] = `
<button
  className="Shorty-Button Shorty-Button--disabled"
  disabled={true}
>
  Lorem ipsum
</button>
`;
